src/logger.rs

```diff
@@ -0,0 +1,48 @@
+// Copyright 2020 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+use crocksdb_ffi;
+use libc::{c_char, c_void};
+use librocksdb_sys::DBLogger;
+use std::ffi::VaList;
+
+pub trait Logger: Send + Sync {
+    fn logv(&self, format: *const c_char, ap: VaList);
+}
+
+#[allow(dead_code)]
+extern "C" fn destructor(ctx: *mut c_void) {
+    unsafe {
+        Box::from_raw(ctx as *mut Box<dyn Logger>);
```

Comment:
add a test similar to this one to make sure the destructor is correct.: https://github.com/tikv/rust-rocksdb/blob/master/src/encryption.rs#L516

Reply:
Done. Add test in the ```cases``` folder.